### DISCUSSION TITLE
[Feature request] `ALTER REL TABLE GROUP`: Add `FROM ... TO ...` statement

### DISCUSSION THREAD

**Discussion by sapalli2989** [kuzudb/kuzu#2929] [Ideas]
It would be nice, if we could add additional node table pairs via `FROM ... TO` *after* creation of a `REL TABLE GROUP` relationship table. For example you need to alter database schema in the course of app progress. Or imagine an app, that allows structured, user-generated content via some kind of plugin system. This requires users to create additional tables. These tables might need to have relations to other tables. Some of these relations might have been instantiated via `REL TABLE GROUP`. Vice versa, same goes for `DROP`. Currently there is no `ALTER` in combination with manipulation of `FROM ... TO` afterwards. At least I couldn't find it in below resources:

https://kuzudb.com/docusaurus/cypher/data-definition/create-table#create-rel-table-group
https://kuzudb.com/docusaurus/cypher/data-definition/alter/

**Comment by sapalli2989** [comment]
One more clarification: Given
```cypher
CREATE REL TABLE GROUP Knows (FROM User To User, FROM User to City, year INT64);
```
What I'd like to have is an `ALTER` command to `ADD` or `DROP` relations from `Knows` at some point later on, similar to
```
ALTER REL TABLE GROUP Knows ADD FROM Person TO Person;
ALTER REL TABLE GROUP Knows DROP FROM User TO User;
```

**Comment by yhtest1** [comment]
Hi, may I know if there is any update yet?

**Comment by ubmarco** [comment]
I'd also be interested in this as I am merging multiple database into new ones over time. Recreating group relationship tables is quite an effort.

**Comment by prrao87** [comment]
Hi @ubmarco and @sapalli2989, this is being worked on actively by @andyfengHKU and @ray6080. I recommend tracking and subscribing to https://github.com/kuzudb/kuzu/issues/3627 to keep updated on the progress.

Basically, we will unify the `CREATE REL TABLE ...` and `CREATE REL TABLE GROUP ...` APIs to only retain the former - this will be a breaking change for users who are using the `GROUP` keyword, but it's just a small syntactic change for a single line in the schema. From a user perspective, we will treat rel table groups the same as single rel tables at the time of creation, and it will become a lot easier to work with them going forward.

**Comment by prrao87** [comment]
I'm closing this discussion as we have #3627 to carry on this conversation.
